Validators/FatturaOrdinariaValidator.cs

```diff
@@ -98,7 +98,8 @@ private static bool BodyValidateAgainstError00444(FatturaElettronicaBody body)
 
             var riepilogo = body.DatiBeniServizi.DatiRiepilogo.Select(x => x.Natura);
```

Comment:
I test falliscono perchè l'array `riepilogo` non è vuoto quando non ci sono campi `DatiRiepilogo.Natura` valorizzati. Se modifichi la linea 99 in questo modo:

```cs
var riepilogo = body.DatiBeniServizi.DatiRiepilogo.Select(x => x.Natura).Where(x => x != null);
```

I test passano tutti.

Reply:
Nicola grazie per il suggerimento ma il problema è proprio questo: se ho una riga di riepilogo senza natura (un elemento dell'array a null) devo necessariamente avere anche una riga di dettaglio senza natura.
Anche il viceversa può generare lo stesso errore quindi non posso escludere i DatiRiepilogo che hanno natura = null, ovvero non posso selezionare solo i DatiRiepilogo con Natura != null perchè ignorerei parte del controllo che lo SDI effettua.
Non so se sono riuscito a spiegarmi

Nella funzione BodyValidateAgainstError00443 la logica di controllo dell'AliquotaIVA è giusta e la stessa cosa deve essere fatta sulla Natura (funzione BodyValidateAgainstError00444)
Sulle specifiche infatti riporta lo stesso tipo di controllo

> - Codice: 00443
Descrizione: non c’è corrispondenza tra i valori indicati nell’elemento 2.2.1.12 <AliquotaIVA> o 2.1.1.7.5 <AliquotaIVA> e quelli dell’elemento 2.2.2.1 <AliquotaIVA> (tutti i valori delle aliquote IVA presenti nelle linee di dettaglio di una fattura o nei dati di cassa previdenziale devono essere presenti anche nei dati di riepilogo)
(vale solo per le fatture ordinarie)

> - Codice: 00444
> Descrizione: non c’è corrispondenza tra i valori indicati nell’elemento 2.2.1.14 <Natura> o 2.1.1.7.7 <Natura> e quelli dell’elemento 2.2.2.2 <Natura> (tutti i valori di natura dell’operazione presenti nelle linee di dettaglio di una fattura o nei dati di cassa previdenziale devono essere presenti anche nei dati di riepilogo)
> (vale solo per le fatture ordinarie)

Ovviamente non è espressamente indicato che è valido anche il viceversa! L'ho dedotto dalle notifiche di scarto che ho ricevuto dallo SDI